mpi/parallel_block_vector_03.mpirun=2.debug fails

This test fails for me:
```
> ctest -V -R mpi/parallel_block_vector_03.mpirun=2.debug
UpdateCTestConfiguration  from :/home/fac/f/bangerth/p/deal.II/1/build/DartConfiguration.tcl
UpdateCTestConfiguration  from :/home/fac/f/bangerth/p/deal.II/1/build/DartConfiguration.tcl
Test project /home/fac/f/bangerth/p/deal.II/1/build
Constructing a list of tests
Done constructing a list of tests
Checking test dependency graph...
Checking test dependency graph end
test 6241
    Start 6241: mpi/parallel_block_vector_03.mpirun=2.debug

6241: Test command: /raid/bangerth/bin/cmake-3.6.1/bin/cmake "-DTRGT=parallel_block_vector_03.mpirun2.debug.diff" "-DTEST=mpi/parallel_block_vector_03.mpirun=2.debug" "-DEXPECT=PASSED" "-DBINARY_DIR=/home/fac/f/bangerth/p/deal.II/1/build/tests/mpi" "-DGUARD_FILE=/home/fac/f/bangerth/p/deal.II/1/build/tests/mpi/parallel_block_vector_03.debug/interrupt_guard.cc" "-P" "/home/fac/f/bangerth/p/deal.II/1/build/share/deal.II/scripts/run_test.cmake"
6241: Test timeout computed to be: 600
6241: gmake[3]: *** [parallel_block_vector_03.debug/mpirun=2/diff] Error 1
6241: gmake[2]: *** [CMakeFiles/parallel_block_vector_03.mpirun2.debug.diff.dir/all] Error 2
6241: gmake[1]: *** [CMakeFiles/parallel_block_vector_03.mpirun2.debug.diff.dir/rule] Error 2
6241: gmake: *** [parallel_block_vector_03.mpirun2.debug.diff] Error 2
6241: Test mpi/parallel_block_vector_03.mpirun=2.debug: DIFF
6241: ===============================   OUTPUT BEGIN  ===============================
6241: Built target parallel_block_vector_03.debug
6241: Generating parallel_block_vector_03.debug/mpirun=2/diff
6241: mpi/parallel_block_vector_03.mpirun=2.debug: BUILD successful.
6241: mpi/parallel_block_vector_03.mpirun=2.debug: RUN successful.
6241: mpi/parallel_block_vector_03.mpirun=2.debug: DIFF failed. ------ Source: /home/fac/f/bangerth/p/deal.II/1/dealii/tests/mpi/parallel_block_vector_03.with_64bit_indices=off.mpirun=2.output
6241: mpi/parallel_block_vector_03.mpirun=2.debug: DIFF failed. ------ Result: /home/fac/f/bangerth/p/deal.II/1/build/tests/mpi/parallel_block_vector_03.debug/mpirun=2/output
6241: Check /home/fac/f/bangerth/p/deal.II/1/build/tests/mpi/parallel_block_vector_03.debug/mpirun=2/output /home/fac/f/bangerth/p/deal.II/1/dealii/tests/mpi/parallel_block_vector_03.with_64bit_indices=off.mpirun=2.output
6241: mpi/parallel_block_vector_03.mpirun=2.debug: DIFF failed. ------ Diff:   /home/fac/f/bangerth/p/deal.II/1/build/tests/mpi/parallel_block_vector_03.debug/mpirun=2/failing_diff
6241: mpi/parallel_block_vector_03.mpirun=2.debug: DIFF failed. ------ First 8 lines of numdiff/diff output:
6241: ----------------
6241: ##3       #:3   <== 599
6241: ##3       #:3   ==> 687
6241: @ Absolute error = 8.8000000000e+1, Relative error = 1.4691151920e-1
6241: ----------------
6241: ##4       #:3   <== 1476
6241: ##4       #:3   ==> 1542
6241: @ Absolute error = 6.6000000000e+1, Relative error = 4.4715447154e-2
6241: mpi/parallel_block_vector_03.mpirun=2.debug: DIFF failed. ------ First 50 lines diff output:
6241: *** /home/fac/f/bangerth/p/deal.II/1/dealii/tests/mpi/parallel_block_vector_03.with_64bit_indices=off.mpirun=2.output     Mon Nov  7 18:33:01 2016
6241: --- output        Thu Apr  6 11:20:23 2017
6241: ***************
6241: *** 1,8 ****
6241:   
6241:   DEAL:0::numproc=2
6241: ! DEAL:0::599
6241: ! DEAL:0::1476
6241:   
6241: ! DEAL:1::519
6241: ! DEAL:1::1254
6241:   
6241: --- 1,8 ----
6241:   
6241:   DEAL:0::numproc=2
6241: ! DEAL:0::687
6241: ! DEAL:0::1542
6241:   
6241: ! DEAL:1::615
6241: ! DEAL:1::1326
6241:   
6241: 
6241: 
6241: mpi/parallel_block_vector_03.mpirun=2.debug: ******    DIFF failed    *******
6241: 
6241: ===============================    OUTPUT END   ===============================
```
Do others see this as well?

--- a/source/base/partitioner.cc
+++ b/source/base/partitioner.cc
@@ -269,7 +269,6 @@ namespace Utilities
 
         // allocate memory for import data
         std::vector<std::pair<unsigned int,unsigned int> > import_targets_temp;
-        import_targets_temp.reserve(n_procs);
         n_import_indices_data = 0;
         for (unsigned int i=0; i<n_procs; i++)
           if (receive_buffer[i] > 0)
@@ -278,8 +277,8 @@ namespace Utilities
               import_targets_temp.push_back(std::pair<unsigned int,
                                             unsigned int> (i, receive_buffer[i]));
             }
-
-        import_targets_data = std::move(import_targets_temp);
+        // copy, don't move, to get deterministic memory usage.
+        import_targets_data = import_targets_temp;
       }
 
       // send and receive indices for import data. non-blocking receives and

